code/modules/automation/auto_chem_filter.dm

```diff
@@ -0,0 +1,72 @@
+//An automatic chem filter; think of it like a automated chem master somewhat
+//Takes a reagent container as a input and will remove chemicals that don't fit with the inputted chem macro
+//Alternatively, the machine can also remove certain reagents VIA chem macro input instead of making reagents like a chem macro
+
+#define FILTER_INTO 0
+#define FILTER_OUT 1
+
+/obj/machinery/automation/chem_filter
```

Comment:
port plumbing and use this instead I feel like this is cringe and just a re-implemented plumbing system while the plumbing system is more fun / cool anyways